src/trie.c

```diff
@@ -84,3 +81,83 @@ int insert_string(char *word, trie_t *t)
         return insert_string(word, t->children[(unsigned)curr]);
     }
 }
+
+
+trie_t *trie_search_end(char* word, trie_t *t)
+{
+
+    int len;
+    trie_t* curr;
+    trie_t** next;
+
+    len = strlen(word);
+    curr = t;
+    next = t->children;
+
+    for (int i=0; i<len; i++) {
```

Comment:
This loop needs documentation. I have absolutely no idea what it's doing - if code takes longer than a glance to understand, there need to be inline comments explaining each step.

Reply:
Will fix in separate issue